### DISCUSSION TITLE
How to update using unique together fields

### DISCUSSION THREAD

**Discussion by shivangrathore** [prisma/prisma#23945] [Q&A] [answered]
### Question

I have a cart model, I want to update it based on userId and productId which are unique together

```prisma
model Cart {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  userId    String   @db.ObjectId
  productId String
  amount    Int
  createdAt DateTime @default(now())
  User      User     @relation(fields: [userId], references: [id])
  @@unique([userId, productId])
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @Wiper-R 👋 

Thank you for raising this question.

To update a `Cart` based on `userId` and `productId` which are unique together, you can use the [`update` ](https://www.prisma.io/docs/orm/prisma-client/queries/crud#update)method. Here is an example of how you can do it:
```js
const updatedCart = await prisma.cart.update({
  where: { 
    userId_productId: {
      userId: user.id,
      productId: product.id
    }
  },
  data: {
    // your update data here
  },
})
```
In the `where` clause, you need to use the unique constraint (`userId_productId`) that you defined in your Cart model. This constraint is a combination of `userId` and `productId`. You replace `user.id` and `product.id` with the actual `userId` and `productId` you want to update.

In the data object, you specify the new values for the fields you want to update.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by shivangrathore** [comment]
Yeah, it works now, Maybe I forgot to save schema file before running `prisma generate`. Thanks though
